tasks/libs/common/git.py

```diff
@@ -270,3 +275,20 @@ def get_last_release_tag(ctx, repo, pattern):
             last_tag_name = last_tag_name_with_suffix.removesuffix("^{}")
     last_tag_name = last_tag_name.removeprefix("refs/tags/")
     return last_tag_commit, last_tag_name
+
+
+def get_git_config(key):
+    result = subprocess.run(['git', 'config', '--get', key], capture_output=True, text=True)
+    return result.stdout.strip() if result.returncode == 0 else None
+
+
+def set_git_config(key, value):
+    subprocess.run(['git', 'config', key, value])
+
+
+def revert_git_config(original_config):
+    for key, value in original_config.items():
+        if value is None:
+            subprocess.run(['git', 'config', '--unset', key])
+        else:
+            subprocess.run(['git', 'config', key, value])
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Could you use `ctx.run` instead? (e.g. `ctx.run('...', hide=True).stdout`). Note that this will raise an error if the process exits with non zero value

Reply:
I'm trying to reuse what the OTel folks did [here](https://github.com/DataDog/datadog-agent/pull/31545/files#diff-909ceda28445b406c806e55409c5e4853ba7f58c2bb4e1b4ef43c3f781a7606dR509), following @chouetz's offline suggestion